Assets/Scripts/Spawn System/SpawnPoint.cs

```diff
@@ -0,0 +1,29 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+
+//one per scene
+public class SpawnPoint : MonoBehaviour
+{
+    PlayerRespawn player;
+
+    public bool hasCheckPoint = false; //has a checkpoint been reached
+    public Vector3 position; //place to spawn character
+    [SerializeField] int height = 1; //how high above repawn to position player
+    // Start is called before the first frame update
+    void Start()
+    {
+        player = GameObject.FindGameObjectWithTag("Player").GetComponent<PlayerRespawn>();
+        position = new Vector3(this.transform.position.x, height,this.transform.position.z);
+
+        // places player at level spawn at start of level 
+          //2nd load onwards
+        player.respawnPoint = position;
+        player.respawnPlayer();
+        
+        
+    }
+
+    
+}
```

Comment:
Probably best to remove this class entirely, and justhave an option to pass an initial checkpoint object into the playerrespawn through a serializefield. I don't think we need the non-revisiting property of this behavior (to my knowledge, the only thing that would functionally change), but if we want that option, adding a toggle for non-revisiting to checkpoint is probably the best way to go about it.

Reply:
(this should also make loading saves easier)